### ISSUE TITLE
rotate page by 90 degree

### ISSUE THREAD

**Issue by kwladyka** [ralfstuckert/pdfbox-layout#2]
I found something like that:
```Java
PageFormat a5_landscape = 
PageFormat.with().mediaBox(
new PDRectangle( Constants.A5.getHeight(),Constants.A5.getWidth()))
.orientation(Orientation.Landscape)
.margins(hMargin, hMargin, 100f, 130f)
.build();
```

but it only change dimension form XxY to YxX. It is worthless. It doesn't rotate the content.

Also i found:
```java
document.getPDDocument().getPage(0).setRotation(90);
```

But is it really no way to set rotation directly on the page?

**Comment by ralfstuckert** [comment]
What orientation() does, is making sure that the resulting width/height matches your desired orientation. Means: if you create a PageFormat with standard A4 - which is portrait since height is bigger than width - and set orientation to landscape, it will be rotated by 90 degrees.

In your example above you create a media box which is already landscape, and set orientation to landscape. So this will have no effect.

So, I'm not quite sure what you want. Could you give me an example and what you expect as a result?

By the way, I found a bug, that if you create a PageFormat with a media box in landcape format, and do not specify an orientation, it will be rotated to portrait, since this is the default orientation. I will open a separate bug for that.

**Comment by kwladyka** [comment]
I am preparing PDF for label printer. This labels are in horizontal orientation in the printer. So i have to rotate each page by 90 degree.

So label has 43x58 [mm] but i want make PDF with page 58x43 and rotated content.

In another world if page is `|` i want to print it as `-` so the dimension and content is rotated.

**Comment by ralfstuckert** [comment]
So if you specify a media box with 43x58 and orientation landscape, does that not give you the desired result?

Or do you want the content (text) to be rotated, is that what you want?

**Comment by ralfstuckert** [comment]
If the latter is the case, I have to tell you that is is not designed for that, since I had no usecase for that (until now).

As a workaround you could add a RenderListener, and rotate the page after rendering. As a result, the content would be rendered horizontally, and than rotated.

**Comment by kwladyka** [comment]
The second one: i need effect `.setRotation 90` on `PDPage`.

I guess this `document.getPDDocument().getPage(0).setRotation(90)` doing the same. So on the end i can do loop `getPage` to rotate pages, but it is not so nice solution :)

**Comment by ralfstuckert** [comment]
So what you actually want - in terms of an API - is a text- resp. content-direction, right?

**Comment by kwladyka** [comment]
```clojure
page (-> (PageFormat/with)
                 (.mediaBox rectangle)
                 ;.landscape
                 (.margins 0 0 0 0)
                 .build)
```

Just wen i creating page it will be nice to have something like `.landspace`, but `(.rotation 90)`. Like in `PDPage`.

PS It is part of code from Clojure.

**Comment by ralfstuckert** [comment]
Added `rotation()` to PageFormat
Fixed in 0.7.2
